### PR TITLE
Adds database provisioning

### PR DESCRIPTION
**Pull request by 97jaz**
Also a bunch of refactoring

### CODE REVIEW

File: lib/aptly/client.rb [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+      def def_collection(collection_name, member_class)
+        define_method(collection_name.to_sym) do
+          get_collection(collection_name, member_class)
+        end
+
+        define_method("find_#{collection_name.chop}".to_sym) do |id|
+          find_collection_member(collection_name, member_class, id)
+        end
+      end
+    end
```
Comment (cjjuice) [inline lib/aptly/client.rb:13]: meta man... m e t a. 👍. nice helpers.

File: spec/aptly/account_spec.rb [hunk 2]
```diff
@@ -2,33 +2,25 @@
...
-    {
-      "id" => 666,
-      "handle" => "account-name",
-      "_links" => {
-        "apps" => {
-          "href" => "https://example.com/accounts/666/apps"
-        }
-      }
-    }
+    JsonFixtures.accounts.last
```
Comment (cjjuice) [inline spec/aptly/account_spec.rb:24]: 👍 much better as a fixture

File: spec/support/json_fixtures.rb [hunk 3]
```diff
@@ -0,0 +1,177 @@
+module JsonFixtures
```
Comment (cjjuice) [inline spec/support/json_fixtures.rb:1]: Much better!

**Review by cjjuice — APPROVED** [review]
LGTM!

**Review by hernanat — APPROVED** [review]
heart eyeballs

**Comment by 97jaz** [comment]
thanks

**Pull request merged**